stanc.ml

```diff
@@ -64,50 +64,49 @@ let options =
       , " Takes a comma-separated list of directories that may contain a file \
          in an #include directive (default = \"\")" ) ]
 
-(** Add a file to the list of files to be loaded, and record whether it should
-      be processed in interactive mode. *)
-let add_file filename = files := filename :: !files
+let model_file_err () =
+  Arg.usage options ("Please specify one model_file.\n\n" ^ usage) ;
+  exit 127
+
+let add_file filename =
+  if !model_file = "" then model_file := filename else model_file_err ()
 
 (** ad directives from the given file. *)
 let use_file filename =
-  let _ =
-    if !Semantic_check.model_name = "" then
-      Semantic_check.model_name :=
-        String.drop_suffix
-          (List.hd_exn (List.rev (String.split filename ~on:'/')))
-          5
-        ^ "_model"
-  in
   let ast =
     try Parse.parse_file Parser.Incremental.program filename
     with Errors.SyntaxError err ->
       Errors.report_syntax_error err ;
       exit 1
   in
   let _ = Debugging.ast_logger ast in
-  let _ = Debugging.auto_formatter ast in
+  if !pretty_print_program then
+    print_endline (Pretty_printing.pretty_print_program ast) ;
   let typed_ast =
     try Semantic_check.semantic_check_program ast
     with Errors.SemanticError err ->
       Errors.report_semantic_error err ;
       exit 1
   in
   let _ = Debugging.typed_ast_logger typed_ast in
-  let mir = Ast_to_Mir.trans_prog filename typed_ast in
-  (*let _ = Stan_math_backend.emit_prog Format.str_formatter mir in
-    let cpp = Format.flush_str_formatter () in
-    print_string cpp
-  *)
-  ignore mir
+  if not !pretty_print_program then (
```

Comment:
Hmm. I think @bob-carpenter expressed a preference for all command line arguments to be independent/orthogonal.
That is, he suggested we introduce an extra command line argument for suppressing the code gen for in case we only want to pretty print and not generate code. That way, we can also pretty print and generate code at the same time.
I don't have any strong feelings about this, myself, however.

Reply:
Ah, I'll have to talk about it with him at some point. I think it makes more sense to think of the compiler as having different actions associated with it. It's unlikely someone wants to generate C++ and pretty print, especially if both are going to stdout or something.